ui/src/lib/access/user.ts

```diff
@@ -28,9 +29,44 @@ export const loadUser = async (casUser: CasUser): Promise<User> => {
     } as User;
 
     await setRoles(user);
+    console.log('CAS user after setRoles:', user);
+    return user;
+};
+
+export const loadOotbUser = async (profile: OotbActiveProfile): Promise<User> => {
+    const user = {
+        name: profile.attributes.cn,
+        firstName: profile.attributes.givenName,
+        lastName: profile.attributes.sn,
+        uid: profile.uid,
+        uhUuid: profile.uhUuid,
+        roles: [Role.ANONYMOUS, ...convertAuthoritiesToRoles(profile.authorities)]
+    } as User;
+
     return user;
 };
 
-export const getUser = async (): Promise<User> => (await getCurrentUser<User>()) ?? AnonymousUser;
+const convertAuthoritiesToRoles = (authorities: string[]): Role[] => {
+    return authorities.map(authority => authority.replace(/^ROLE_/, '').toUpperCase())
+                      .filter(roleName => Object.values(Role).includes(roleName as Role))
+                      .map(roleName => roleName as Role);
+};
+
+
+export const getUser = async (): Promise<User> => {
+    if (process.env.NEXT_PUBLIC_OOTB_MODE === 'true') {
+        const givenName = process.env.NEXT_PUBLIC_OOTB_PROFILE;
+        try {
+            await updateActiveDefaultUser(givenName);
+            const profile = await matchProfile(givenName);
+            return profile ? await loadOotbUser(profile) : AnonymousUser;
+        } catch (error) {
+            console.error('Error fetching OOTB user:', error);
+            return AnonymousUser;
+        }
```

Comment:
Separate into new function `const getOotbUser` that gets called in this if statement.